src/map/script.c

```diff
@@ -16473,21 +16473,28 @@ static BUILDIN(unequip)
 
 static BUILDIN(equip)
 {
-	int nameid=0,i;
+	int nameid = 0, i;
 	struct item_data *item_data;
 	struct map_session_data *sd = script->rid2sd(st);
-	if (sd == NULL)
-		return false;
 
-	nameid=script_getnum(st,2);
-	if((item_data = itemdb->exists(nameid)) == NULL)
-	{
-		ShowError("wrong item ID : equipitem(%d)\n",nameid);
-		return false;
+	if (sd == NULL) {
+		script_pushint(st, 0);
+		return true;
+	}
+
+	nameid = script_getnum(st, 2);
+	if ((item_data = itemdb->exists(nameid)) == NULL) {
+		ShowError("buildin_equip: Invalid Item ID (%d).\n",nameid);
+		script_pushint(st, 0);
+		return true;
```

Comment:
`return false;`

the condition above, (sd == NULL) has to `return true;` because `script->rid2sd` has `script->reportsrc(st);` in it